ported/librdmacm/cmatose.c

```diff
@@ -354,7 +357,10 @@ static int connect_events(void)
 
 	while (connects_left && !ret) {
 		ret = fi_eq_sread(eq, &event, &entry, sizeof entry, -1, 0);
-		if (ret < 0) {
+		
+		if (ret == -FI_EAGAIN)
+			continue;
+		else if (ret < 0) {
```

Comment:
I would change the loop here as well to check for ret == -FI_EAGAIN and modify the if statement inside the loop to check for ret < 0 and ret != -FI_EAGAIN